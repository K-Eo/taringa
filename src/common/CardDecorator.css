.Card {
  position: relative;
}

.card.link:hover {
  background-color: #f8f9fa;
  cursor: pointer;
  text-decoration: none;
}

.Card .card-body {
  padding-top: 2.2rem !important;
}

.Card .card-title {
  font-size: 1.2rem;
}

.Card .card-subtitle {
  font-size: .9rem;
}

.Card-cover {
  height: 100px;
  max-height: 100px;
  max-width: 100%;
  min-width: 100%;
  min-width: 100px;
  object-fit: cover;
  width: 100%;
}

.Card-avatar {
  position: absolute;
  top: calc(100px - (64px - 24px));
  left: 1.25rem;
  width: 64px;
  height: 64px;
  background: #dee2e6;
  box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.3);
}

@keyframes cardPlaceholderShimmer{
  0%{
      background-position: -348px 0
  }
  100%{
      background-position: 348px 0
  }
}

.Card-animated-background {
  animation-duration: 1s;
  animation-fill-mode: forwards;
  animation-iteration-count: infinite;
  animation-name: cardPlaceholderShimmer;
  animation-timing-function: linear;
  background: #f6f7f8;
  background: linear-gradient(to right, #eeeeee 8%, #dddddd 18%, #eeeeee 33%);
  background-size: 688px 340px;
  height: 300px;
  position: relative;
  border-radius: 0.25rem;
}

.Card-ph-avatar {
  position: absolute;
  top: calc(100px - (64px - 24px));
  left: 1.25rem;
  width: 64px;
  height: 64px;
  background: #dee2e6;
}

.Card-ph-body {
  position: absolute;
  top: 100px;
  bottom: 0;
  left: 0;
  width: 100%;
  background: white;
  border-radius: 0 0 0.25rem 0.25rem;
}
