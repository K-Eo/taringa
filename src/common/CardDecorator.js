import React from 'react'
import classNames from 'classnames'
import LazyLoad from 'react-lazyload'

import './CardDecorator.css'

import history from '../history'

const CardDecorator = ({
  avatar,
  children,
  className,
  cover,
  link,
  placeholder,
  to,
  ...rest
}) => {
  if (placeholder) {
    return (
      <div className="Card card">
        <div className="Card-animated-background">
          <div className="Card-ph-body" />
          <div className="Card-ph-avatar rounded" />
        </div>
      </div>
    )
  }

  const handleOnClick = () => {
    if (link && to) {
      history.push(to)
    }
  }

  const classes = classNames('Card card', className, {
    link: link && true,
  })

  return (
    <div {...rest} onClick={handleOnClick} className={classes}>
      {avatar && (
        <LazyLoad placeholder={<div className="Card-avatar" />} debounce once>
          <img src={avatar} className="Card-avatar rounded" alt="Avatar" />
        </LazyLoad>
      )}

      <LazyLoad height={100} debounce once>
        <img
          alt="Cover"
          className="Card-cover card-img-top"
          src={cover || 'https://k60.kn3.net/taringa/0/C/B/5/A/D/9AA.png'}
        />
      </LazyLoad>
      {children}
    </div>
  )
}

CardDecorator.defaultProps = {
  placeholder: false,
}

export default CardDecorator
