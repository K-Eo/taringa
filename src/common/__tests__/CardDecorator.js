import React from 'react'

import CardDecorator from '../CardDecorator'

describe('CardDecorator', () => {
  it('exists', () => {
    expect(CardDecorator).toBeDefined()
  })

  it('renders', () => {
    const wrapper = shallow(<CardDecorator />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with custom params', () => {
    const wrapper = shallow(
      <CardDecorator
        avatar="https://placehold.it/64"
        className="foo"
        cover="https://placehold.it/600x200"
        placeholder={false}
      >
        bar
      </CardDecorator>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
