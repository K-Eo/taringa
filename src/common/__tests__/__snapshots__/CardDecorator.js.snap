// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardDecorator renders 1`] = `
<div
  className="Card card"
>
  <div
    className="Card-animated-background"
  >
    <div
      className="Card-ph-body"
    />
    <div
      className="Card-ph-avatar rounded"
    />
  </div>
</div>
`;

exports[`CardDecorator renders as a link 1`] = `
<div
  className="Card card link"
  onClick={[Function]}
>
  <LazyLoad
    height={64}
    offset={100}
    once={true}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <img
      alt="Avatar"
      className="Card-avatar rounded"
      src="https://placehold.it/64"
    />
  </LazyLoad>
  <LazyLoad
    height={100}
    offset={100}
    once={true}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <img
      alt="Cover"
      className="Card-cover card-img-top"
      src="https://placehold.it/600x200"
    />
  </LazyLoad>
  bar
</div>
`;

exports[`CardDecorator renders with custom params 1`] = `
<div
  className="Card card foo"
  onClick={[Function]}
>
  <LazyLoad
    height={64}
    offset={100}
    once={true}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <img
      alt="Avatar"
      className="Card-avatar rounded"
      src="https://placehold.it/64"
    />
  </LazyLoad>
  <LazyLoad
    height={100}
    offset={100}
    once={true}
    overflow={false}
    resize={false}
    scroll={true}
    unmountIfInvisible={false}
  >
    <img
      alt="Cover"
      className="Card-cover card-img-top"
      src="https://placehold.it/600x200"
    />
  </LazyLoad>
  bar
</div>
`;
